### PR TITLE
feat: coverage-collector-implementation

### PR DESCRIPTION
**Pull request by alex-paca-jalafund**
#### Resolved task/US: ([CS-005-1](https://app.clickup.com/t/868azy24a))
* * * *

## What's in this pull request?

In this PR its the implementation of coverage collector, editing the injecter for comunicate with the path ov the coverageCOllector collect method also updating the corresponding test.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Its implemented unit testing for this PR, simulating a valid and not valid collection. Also this PR will be tested for a end to end testing at the end of the complete testing
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosInstrumentation/11)

### CODE REVIEW

File: src/main/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecter.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public void visitLineNumber(int line, Label start) {
         mv.visitIntInsn(SIPUSH, line);
         mv.visitMethodInsn(
                 INVOKESTATIC,
-                "driver/CoverageCollect",
+                "edu/usb/argosinstrumentation/transformer/CoverageCollector",
```
Comment (zamoranoalicia) [inline src/main/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecter.java]: please use a constant here, you can add a new field on the application properties file and read it using @Value from spring boot

File: src/test/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecterTest.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public void visitMethodInsn(
                                         String descriptor,
                                         boolean isInterface) {
                                     assertEquals(
-                                            "driver/CoverageCollect",
+                                            "edu/usb/argosinstrumentation/transformer/CoverageCollector",
```
Comment (zamoranoalicia) [inline src/test/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecterTest.java]: same here

**Review by zamoranoalicia — CHANGES_REQUESTED** [review]

File: src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import edu.usb.argosinstrumentation.domain.CoverageData;
+import edu.usb.argosinstrumentation.domain.MethodData;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CoverageCollector {
+
+    private static Map<String, CoverageData> finalInfo = new HashMap<>();
+
+    public CoverageCollector(Map<String, CoverageData> finalInfo) {
```
Comment (SebasthianSalpozz) [inline src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java:17]: This constructor should be declared with lombok I think.

**Review by SebasthianSalpozz — CHANGES_REQUESTED** [review]

**Review by zrock-dev — CHANGES_REQUESTED** [review]
Good job. I've requested some changes. 

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosInstrumentation/11#-:-OCfBqnK7LwNtipQaec7:bl9ye2l)* status: all files reviewed, 6 unresolved discussions (waiting on @alex-paca-jalafund and @bianca-soliz-jalafund)

___
*[`src/main/java/edu/usb/argosinstrumentation/domain/MethodData.java` line 7 at r1](https://reviewable.io/reviews/SoftDevSix/ArgosInstrumentation/11#-OCf9SFT0Vp3_dqM-PIY:-OCf9SFT0Vp3_dqM-PIZ:b-apx0kx) ([raw file](https://github.com/SoftDevSix/ArgosInstrumentation/blob/087ed25d5a39dfb2e9228b9ce69cc4748730268d/src/main/java/edu/usb/argosinstrumentation/domain/MethodData.java#L7)):*
> ```Java
> import lombok.Getter;
> 
> @AllArgsConstructor
> ```

Please clarify why this annotation is required.
___
*[`src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java` line 18 at r1](https://reviewable.io/reviews/SoftDevSix/ArgosInstrumentation/11#-OCfB9dV1lGrY69ejzlb:-OCfB9dV1lGrY69ejzlc:b-uw61nn) ([raw file](https://github.com/SoftDevSix/ArgosInstrumentation/blob/087ed25d5a39dfb2e9228b9ce69cc4748730268d/src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java#L18)):*
> ```Java
>     public static void collect(String className, String methodName, String mDesc, int line) {
>         if (finalInfo.get(className) != null) {
>             MethodData methodData = new MethodData(methodName, mDesc);
> ```

Please use MethodData's builder instead, this way if the class implementation changes this code won't be affected.
___
*[`src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java` line 21 at r1](https://reviewable.io/reviews/SoftDevSix/ArgosInstrumentation/11#-OCfA1VKA4lVWt5xkq9N:-OCfA1VKA4lVWt5xkq9O:bymwdu1) ([raw file](https://github.com/SoftDevSix/ArgosInstrumentation/blob/087ed25d5a39dfb2e9228b9ce69cc4748730268d/src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java#L21)):*
> ```Java
>             finalInfo.get(className).getClassData().saveMethodData(methodData, line);
>         } else {
>             System.out.println("final info get className is null at collectMethod");
> ```

Please use a logger instead. This way we will avoid losing important log data.

File: src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java (hunk 3 above)
Comment (alex-paca-jalafund) [reply src/main/java/edu/usb/argosinstrumentation/transformer/CoverageCollector.java:17]: Lombok dosn't detect static implementations such us finalInfo, so when i used allArgsConstructor or Builder generates an empty builder, in order to have separate classes i create a manual construct in orther to set the finalInfo during the process.

File: src/main/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecter.java (hunk 1 above)
Comment (alex-paca-jalafund) [reply src/main/java/edu/usb/argosinstrumentation/coverageadapter/CodeInjecter.java]: Miss i run into a problem that if i use @Value directly in codeInjecter for some reason its always null but I implemented in the constructor and the class were is going to be used CoverageAdapter Its passing the @Value, when i  do the same implementation in CodeInjecter class SpringBoot trhows an error that says that there is an cyclic call

**Review by zamoranoalicia — APPROVED** [review]
LGTM

**Pull request closed without merging**